### PR TITLE
WIP: general ui improvements and authentication changes.

### PR DESCRIPTION
**Pull request by Tomasu**
Hi,

I've made a few changes that (imo) improve the ui flow a bit and fix some minor theming issues, and made it authenticate using the nextcloud app. 

This is not a formal pull request as its still quite messy and can't yet fall back to another form of authentication. I'm planning on adding Oauth support.

I'm mostly just making this PR to announce that it exists and requesting comments and feedback.

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -1,110 +1,149 @@
...
+	implementation "androidx.recyclerview:recyclerview:1.0.0"
+	implementation "androidx.cardview:cardview:1.0.0"
+	implementation "androidx.constraintlayout:constraintlayout:1.1.3"
+	implementation "com.google.android.material:material:1.0.0"
+
+	implementation 'com.jakewharton:butterknife:10.1.0'
+	implementation 'com.koushikdutta.ion:ion:2.2.1'
+	implementation 'commons-codec:commons-codec:1.10'
+
+	implementation 'com.google.android.gms:play-services-base:16.1.0'
```
Comment (animalillo) [inline app/build.gradle]: We do not want to depend on any google specific services or anything preinstalled on the phone except android itself

Comment (animalillo) [reply app/build.gradle]: We want the app to be fully compatible with any AOSP phones out there

**Comment by animalillo** [comment]
I've added that tiny comment as I've seen gms referenced, I hope I get some time to properly look into your merge request as it seems to have a real lot of changes and looks very interesting indeed.

File: app/build.gradle (hunk 1 above)
Comment (Tomasu) [reply app/build.gradle]: Ok, I believe I only added that to play with AccountPicker which is uneccesary anyhow.

**Comment by Tomasu** [comment]
> I've added that tiny comment as I've seen gms referenced, I hope I get some time to properly look into your merge request as it seems to have a real lot of changes and looks very interesting indeed.

I fully expect to be doing a bunch more. I will try to keep changes in more descreet commits, but sometimes it's a bit difficult when working through issues. I'm planning on doing _some_ refactoring to more modern android practices like the jetpack architecture and ui components.

**Comment by animalillo** [comment]
@Tomasu if you wish, we can discuss plans and changes on telegram that is the main platform we've been using for communication, poke me at https://t.me/passman_general and i will make sure to add you on the proper dev groups

**Comment by Tomasu** [comment]
> @Tomasu if you wish, we can discuss plans and changes on telegram that is the main platform we've been using for communication, poke me at https://t.me/passman_general and i will make sure to add you on the proper dev groups

Who are you on telegram? Just signed up and I'm in passman_general now. first time I've used telegram.

**Comment by binsky08** [comment]
slightly outdated; close draft because there is no current activity

**Pull request closed without merging**